acq4/devices/FilterWheel/FilterWheel.py

```diff
@@ -0,0 +1,402 @@
+    
+    sigFilterWheelPositionChanged = QtCore.Signal(object)
+    sigFilterWheelSpeedChanged = QtCore.Signal(object)
+    sigFilterWheelTrigModeChanged = QtCore.Signal(object)
+    sigFilterWheelSensorModeChanged = QtCore.Signal(object)
+    
+    def __init__(self, dm, config, name):
+        
+        Device.__init__(self, dm, config, name)
+        OptomechDevice.__init__(self, dm, config, name)
+        
+        self.scopeDev = None
+        self.currentFilter = None
+        p = self
+        while p is not None:
+            p = p.parentDevice()
+            if isinstance(p, Microscope):
+                self.scopeDev = p
+                
+        self.port = config['port']-1  ## windows com ports start at COM1, pyserial ports start at 0
+        self.baud = config.get('baud', 115200)
+        #self.positionLabels = config.get('postionLabels')
+        
+        
+        self.driver = filterWheelDriver(self.port, self.baud)
+        self.driverLock = Mutex(QtCore.QMutex.Recursive)  ## access to low level driver calls
+        self.filterWheelLock = Mutex(QtCore.QMutex.Recursive)  ## access to self.attributes
+        
+        
+        self.filters = collections.OrderedDict()
+        ## Format of self.filters is:
+        ## { 
+        ##    filterWheelPosition1: {filterName: filter},
+        ##    filterWheelPosition2: {filterName: filter},
+        ## }
+        nPos = self.getPositionCount()
+        for k in range(nPos):  ## Set value for each filter
+            filt = Filter(config['filters'],self,k)
+            self.filters[k] = filt
+        
+        #print self.filters.values()
+        #print self.filters[1].name(), self.filters[1].description()
+        #if len(self.positionLabels) != self.getPositionCount():
+        #    raise Exception("Number of FilterWheel positions %s must correspond to number of labels!" % self.getPositionCount())
+        
+        with self.driverLock:
+            self.currentFWPosition = self.driver.getPos()
+            self.currentObjective = self.getFilter()
+            
+        self.fwThread = FilterWheelThread(self, self.driver, self.driverLock)
+        self.fwThread.fwPosChanged.connect(self.positionChanged)
+        self.fwThread.start()
+        
+    def setTriggerMode(self, trigMode):
+        with self.driverLock:
+            self.driver.setTriggerMode(trigMode)
+            self.sigFilterWheelTrigModeChanged.emit(trigMode)
+    
+    def getTriggerMode(self):
+        with self.driverLock:
+            return self.driver.getTriggerMode()
+       
+    def setSpeed(self, speed):
+        with self.driverLock:
+            self.driver.setSpeed(speed)
+            self.sigFilterWheelSpeedChanged.emit(speed)
+    
+    def getSpeed(self):
+        with self.driverLock:
+            return self.driver.getSpeed()
+        
+    def setPosition(self, pos):
+        with self.driverLock:
+            self.driver.setPos(pos)
+            self.sigFilterWheelPositionChanged.emit(pos)
```

Comment:
It seems like this will cause `sigFilterWheelPositionChanged` to be emitted twice--this one is emitted before the move, and then the thread will detect a position change and cause another to be emitted via `positionChanged`; is that correct?